### PR TITLE
Browse by specialty

### PR DESCRIPTION
**Pull request by niquerio**
Added browse by specialty for both peers and candidates. It:
* provides links for specialties
* index specialties shows list of all specialties for given peerage (where there are peers or candidates with those specialties)
* show lists all peers or candidates with a given specialty
* added a view helper that spits out links to specialty page
* removed show_specialties method from peer and candidate

Turned table in groups show and specialties show into shared partial

### CODE REVIEW

File: app/models/candidate.rb [hunk 1]
```diff
@@ -24,12 +24,16 @@ def set_defaults
...
 
-  def show_specialties
-    return self.specialties.map{|s| s.name}.to_sentence
-  end
-
   def order
     peerage_type.to_sym
   end
 
+  def list
```
Comment (aelkiss) [inline app/models/candidate.rb]: How is this change related to specialties?

**Comment by aelkiss** [comment]
Re: view helpers - there's certainly a lot in common between the peers table and candidates table. I think it's OK to wait until either you need to a third similar thing or when you want to make the same change to both the peers and candidates table to try to merge them. Although they are pretty similar, they're still different enough that I think merging the commons stuff between them might make it significantly harder to understand what's going on.

File: app/models/candidate.rb (hunk 1 above)
Comment (niquerio) [reply app/models/candidate.rb]: It's in the specialties and groups candidate table.

File: app/controllers/specialties_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class SpecialtiesController < ApplicationController
+  helper_method :peerage
+  def index
+    @specialties = Specialty.where(peerage_type: peerage)
+  end
+  def show
+    @specialty = Specialty.find_by(peerage_type: peerage, slug: params[:slug])
+  end
+  def peerage
+    params[:peerage]
```
[the remark this answers is not in the document]
Comment (niquerio) [reply app/controllers/specialties_controller.rb]: It should be to_sym in both.

File: app/controllers/chambers/specialties_controller.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+class Chambers::SpecialtiesController < ApplicationController
```
[the remark this answers is not in the document]
Comment (niquerio) [reply app/controllers/chambers/specialties_controller.rb]: done

**Comment by niquerio** [comment]
Re: tests for view_helpers and peers_for: I've added these. I didn't do it before because my integration tests would break if they weren't working, I think, but testing them directly is better.
